modules/javafx.graphics/src/main/java/javafx/concurrent/Task.java

```diff
@@ -1014,12 +1016,31 @@ protected void failed() { }
             // state flag will not be readable immediately after this call. However,
             // that would be the case anyway since these properties are not thread-safe.
             if (isFxApplicationThread()) {
+                switch (getState()) {
+                case FAILED:
+                case SUCCEEDED:
```

Comment:
Avoiding transitioning from a one of the completion states to canceled is a good thing. My question, though is: how did we get here? Is this masking some other problem?

Reply:
I think the variations in the test timing uncovered this issue.  I don't think it's a problem with the execution of the task, but rather with the reporting of its final state.